packages/dds/tree/README.md

```diff
@@ -57,3 +57,98 @@ which could make their way back into the framework, enabling some features speci
+
+## Architecture
+
+This section covers the internal structure the Tree DDS.
+In this section the user of this package is called "the application".
+"The application" is full of "application code", meaning code which can be specific to particular schema and use-cases.
+This typically means the client side "business logic" or "view" part of some graphical web application, but it could also something headless like a service.
+
+### Ownership and Lifetimes
+
+TODO: add a diagram for this section.
+
+`tree` is a DDS, and therefore is stores its persisted data in a Fluid Container, and is also owned by that same container.
+When nothing in that container references the DDS anymore, it may get garbage collected by the Fluid GC.
+
+The tree DDS itself, or more specifically [`shared-tree-core`](./src/shared-tree-core/README.md) is composed of a collection of indexes (just like a database) which contribute data which gets persisted as part of the summary in the container.
+`shared-tree-core` owns these databases, and is responsible for populating them from summaries and updating them when summarizing.
+
+TODO: When support for multiple branches is added, do we want to have indexes for each branch, and if so, maybe their ownership should move to a branch specific structure (like checkout?).
+
+When applications want access to the `tree`'s data, they do so through a [`checkout`](./src/checkout/README.md) which abstracts the indexes into nice application facing APIs.
+Checkouts may also have state from the application, including:
+
+-   [`view-schema`](./src/schema-view/README.md)
+-   adapters for out-of-schema data
+-   request or hints for what subsets of the tree in memory
+-   pending transactions
+-   registrations for application callbacks / events.
+
+[`shared-tree`](./src/shared-tree/) provides a default checkout which it owns, but applications can create more if desired, which they will own.
+Since checkouts subscribe to events from `shared-tree`, explicitly disposing any additionally created ones of is required to avoid leaks.
+
+[transactions](./src/transaction/README.md) are created from `checkouts` and are currently synchronous.
+Support for asynchronous transactions, with the application managing the lifetime and ensuring it does not exceed the lifetime of the checkout,
+could be added in the future.
+
+### Data Flow
+
+#### Viewing
+
+TODO: add a diagram for this section.
+
+[`shared-tree`](./src/shared-tree/) configures [`shared-tree-core`](./src/shared-tree-core/README.md) with a set of indexes.
+`shared-tree-core` downloads the summary data from the Fluid Container, feeding the summary data (and any future edits) into the indexes.
+`shared-tree` then construct's the default `checkout`.
+The application using the `shared-tree` to get the checkout from which it can read data (which the checkout internally gets from the indexes).
```

Comment:
```suggestion
The application using the `shared-tree` can get the checkout from which it can read data (which the checkout internally gets from the indexes).
```